File: mobile/lib/core/config/tarif_config.dart
/// Tarif Jalur A.
///
/// PERINGATAN: seluruh angka di bawah ini masih **placeholder**. Rencana
/// capstone bagian 14.8 menandai "tarif persis Jalur A" sebagai pertanyaan
/// pengunci yang jawabannya harus datang dari mitra:
///
///   - Anter jemput: flat, per kilometer, atau per zona?
///   - Jastip: fee tetap atau persentase nilai belanja?
///
/// Begitu mitra menjawab, cukup ubah angka (atau bentuk rumusnya) di file ini
/// saja — tidak ada harga yang boleh ditulis tersebar di layar mana pun.
library;

class TarifConfig {
  const TarifConfig._();

  // --- Anter Jemput (placeholder: flat + per kilometer) ---
  static const int anjemTarifDasar = 5000;
  static const int anjemTarifPerKm = 2000;
  static const double anjemJarakMinimalKm = 0.5;
  static const double anjemJarakMaksimalKm = 15;

  // --- Jastip Makanan (placeholder: fee tetap, harga barang dibayar terpisah) ---
  static const int jastipMakananFee = 8000;

  // --- Jastip Barang (placeholder: fee tetap + ongkos jarak) ---
  static const int jastipBarangFee = 10000;
  static const int jastipBarangTarifPerKm = 2000;

  /// Batas waktu klien menyelesaikan pembayaran sebelum order hangus.
  /// Angka ini juga menunggu keputusan mitra (bagian 14.7e).
  static const Duration batasWaktuBayar = Duration(minutes: 30);
}

File: backend/src/UpnvjSuruh.Api/Pricing/TarifConfig.cs
namespace UpnvjSuruh.Api.Pricing;

/// <summary>
/// Tarif Jalur A. Kembaran <c>lib/core/config/tarif_config.dart</c> di aplikasi mobile,
/// dan angkanya wajib sama, dijaga tes.
///
/// PERINGATAN: seluruh angka ini masih placeholder. Rencana capstone bagian 14.8 menandai
/// tarif persis Jalur A sebagai pertanyaan pengunci yang jawabannya harus datang dari mitra.
/// </summary>
public static class TarifConfig
{
    public const decimal AnjemTarifDasar = 5000m;
    public const decimal AnjemTarifPerKm = 2000m;
    public const double AnjemJarakMinimalKm = 0.5;
    public const double AnjemJarakMaksimalKm = 15;

    public const decimal JastipMakananFee = 8000m;

    public const decimal JastipBarangFee = 10000m;
    public const decimal JastipBarangTarifPerKm = 2000m;

    public static readonly TimeSpan BatasWaktuBayar = TimeSpan.FromMinutes(30);
}
